### PR TITLE
python3Packages.elastic-transport: init at 8.4.0; python3Packages.elasticsearch: 7.16.3 -> 8.4.2

### PR DESCRIPTION
**Pull request by mweinelt**
###### Description of changes

Updating this package after #192268 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mweinelt** [comment]
Breaks elasticsearch-dsl, which is stuck in the 7.x world.

**Comment by applePrincess** [comment]
Result of `nixpkgs-review pr 192340` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages failed to build:</summary>
  <ul>
    <li>elasticsearch-curator</li>
    <li>mailman-web</li>
    <li>parsedmarc (python310Packages.parsedmarc)</li>
    <li>python310Packages.elasticsearchdsl (python310Packages.elasticsearch-dsl)</li>
    <li>python310Packages.flask_elastic</li>
    <li>python39Packages.elasticsearchdsl (python39Packages.elasticsearch-dsl)</li>
    <li>python39Packages.flask_elastic</li>
    <li>python39Packages.parsedmarc</li>
  </ul>
</details>
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>archivy</li>
    <li>python310Packages.elastic-transport</li>
    <li>python310Packages.elasticsearch</li>
    <li>python39Packages.elastic-transport</li>
    <li>python39Packages.elasticsearch</li>
  </ul>
</details>

**Comment by applePrincess** [comment]
Ah yeah, do we maintain two versions?

File: pkgs/development/python-modules/elasticsearch/default.nix [hunk 1]
```diff
@@ -1,29 +1,78 @@
...
-  propagatedBuildInputs = [ urllib3 requests ];
-  buildInputs = [ nosexcover mock ];
+  checkInputs = [
+    numpy
+    pandas
+    pytest-asyncio
+    pytestCheckHook
+    python-dateutil
+    pyyaml
+  ] ++ lib.flatten (builtins.attrValues passthru.optional-dependencies);
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/elasticsearch/default.nix:69]:
```suggestion
  ] ++ lib.flatten (lib.attrValues passthru.optional-dependencies);
```

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/development/python-modules/elasticsearch/default.nix (hunk 1 above)
Comment (mweinelt) [reply pkgs/development/python-modules/elasticsearch/default.nix:69]: Why do we prefer lib over builtins?

Comment (SuperSandro2000) [reply pkgs/development/python-modules/elasticsearch/default.nix:69]: There is some fallback code in lib https://github.com/NixOS/nixpkgs/blob/master/lib/attrsets.nix#L194

Comment (mweinelt) [reply pkgs/development/python-modules/elasticsearch/default.nix:69]: Thanks.

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by applePrincess** [comment]
https://github.com/elastic/elasticsearch-dsl-py/issues/1569 tracks migration for elasticsearch-dsl from 7.x to 8.x

**Pull request closed without merging**